src/back/spv/writer.rs

```diff
@@ -103,17 +95,13 @@ impl Writer {
         }
     }
 
-    fn get_type_id(
-        &mut self,
-        arena: &crate::Arena<crate::Type>,
-        handle: crate::Handle<crate::Type>,
-    ) -> Word {
-        match self.lookup_type.lookup_id(handle) {
-            Some(word) => word,
-            None => {
-                let (instruction, id) = self.parse_type_declaration(arena, handle);
-                instruction.to_words(&mut self.logical_layout.declarations);
-                id
+    fn get_type_id(&mut self, arena: &crate::Arena<crate::Type>, lookup_ty: LookupType) -> Word {
+        if let Entry::Occupied(e) = self.lookup_type.entry(lookup_ty) {
+            *e.get()
+        } else {
+            match lookup_ty {
```

Comment:
when we add a new type, we have an ID. So we should write down it in the map, i.e.
```rust
match self.lookup_type(lookup_ty) {
  Entry::Occupied(e) => e.get(),
  Entry::Vacant(e) => {
    let id = ...;
    *e.insert(id)
  }
}
```